### ISSUE TITLE
com.axway.apim.api.export.ExportAPI.getCustomProperties  java.lang.NullPointerException

### ISSUE THREAD

**Issue by dnplus** [Axway-API-Management-Plus/apim-cli#108]
**API-Manager CLI Version**
1.2.3

**API-Manager and Service-Pack Version**
7.7.20200130

**Question**
We have manually added some APIs in API Manager, we try to export it in JSON format, we got a null pointer exception at
com.axway.apim.api.export.ExportAPI.getCustomProperties

Actually, we didn't use any custom properties on APIs, not sure if we made any mistake on the cli?
```
API-Manager CLI: 1.2.3

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
----------------------------------------------------------------------------------------
Module: API - E X P O R T / U T I L S (1.2.3)
----------------------------------------------------------------------------------------
573    INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
752    INFO          APIExportApp| 1 API(s) selected.
+--------------------------------------+----------------------+------------+---------+
| API-Id                               | Path                 | Name       | Version |
+--------------------------------------+----------------------+------------+---------+
| 8009cbfe-efd4-45d3-9950-10562de87c5f | /api/oauth/integrate | OAuth整合API |     1.0 |
+--------------------------------------+----------------------+------------+---------+

A P I  -  D E T A I L S
Organization:            API Development
Created On:              Tue Jun 09 15:51:34 CST 2020
Created By:              API Administrator
Granted Organizations:   API Development
Subscribed applications: 絲綢記帳
Custom-Policies:         {FAULT_HANDLER=[], REQUEST=[], RESPONSE=[], ROUTING=[]}
Tags:                    GSS: TWOB
Custom-Properties:       N/A
```

```
API-Manager CLI: 1.2.3

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
----------------------------------------------------------------------------------------
Module: API - E X P O R T / U T I L S (1.2.3)
----------------------------------------------------------------------------------------
757    INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
1116   INFO          APIExportApp| 1 API(s) selected.
1117   INFO       JsonAPIExporter| Going to export API into folder: ./api-oauth-integrate
1140   ERROR      JsonAPIExporter| Can't export API: Can't create API-Configuration file for API: 'OAuth整合API' exposed on path: '/api/oauth/integrate'. Please check in API-Manager UI the API is valid.
com.axway.apim.lib.errorHandling.AppException: Can't create API-Configuration file for API: 'OAuth整合API' exposed on path: '/api/oauth/integrate'.
	at com.axway.apim.api.export.impl.JsonAPIExporter.saveAPILocally(JsonAPIExporter.java:116)
	at com.axway.apim.api.export.impl.JsonAPIExporter.execute(JsonAPIExporter.java:52)
	at com.axway.apim.APIExportApp.execute(APIExportApp.java:154)
	at com.axway.apim.APIExportApp.exportAPI(APIExportApp.java:67)
	at com.axway.apim.APIExportApp.exportAPI(APIExportApp.java:56)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.axway.apim.cli.APIManagerCLI.run(APIManagerCLI.java:142)
	at com.axway.apim.cli.APIManagerCLI.main(APIManagerCLI.java:78)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: (was java.lang.NullPointerException) (through reference chain: com.axway.apim.api.export.ExportAPI["customProperties"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353)
	at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:316)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:727)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:155)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper._configAndWriteValue(ObjectMapper.java:3906)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3160)
	at com.axway.apim.api.export.impl.JsonAPIExporter.saveAPILocally(JsonAPIExporter.java:114)
	... 10 more
Caused by: java.lang.NullPointerException
	at com.axway.apim.api.export.ExportAPI.getCustomProperties(ExportAPI.java:212)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:719)
	... 16 more
```

**What I've tried so far**
We also test on 20200730, and several APIs, the same result.

**Additional information**

**Comment by dnplus** [comment]
Not sure customProperties will exists?
https://github.com/Axway-API-Management-Plus/apim-cli/blob/4437303dc969678e7de8e2b2dad45b2782e55d14/modules/apim-adapter/src/main/java/com/axway/apim/adapter/apis/APIFilter.java#L418

**Comment by cwiechmann** [comment]
Hi @dnplus, this is a known issue in version 1.2.3 and will be fixed in 1.3.0. You can download a SNAPSHOT-Release of 1.3.0 attached to the last comment #84 which already contains the fix. Let me know, if this works for you.

**Comment by cwiechmann** [comment]
Closing this issue, as it is already fixed.
